### PR TITLE
[Merged by Bors] - feat: Complex `exp`, `log`, and `cpow` are analytic

### PR DESCRIPTION
**Pull request by girving**
These are all immediate from differentiability.

We also record several rewrite lemmas for switching between analyticity and differentiability.  These are immediate from existing theorems, but as far as I can tell don't yet exist as iffs.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/Complex/Analytic.lean [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+variable {E : Type} [NormedAddCommGroup E] [NormedSpace ℂ E]
+variable {f g : E → ℂ} {z : ℂ} {x : E} {s : Set E}
+
+/-- `exp` is entire -/
+theorem AnalyticOn.cexp : AnalyticOn ℂ exp univ := by
+  rw [analyticOn_univ_iff_differentiable]; exact differentiable_exp
+
+/-- `exp` is analytic at any point -/
+theorem AnalyticAt.cexp : AnalyticAt ℂ exp z :=
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Complex/Analytic.lean:28]: could you instead call this `analyticAt_cexp` and have a theorem `AnalyticAt.cexp` for dot notation, similar to `AnalyticAt.clog` below?

**Review by sgouezel — COMMENTED** [review]
LGTM, thanks!
bors d+

File: Mathlib/Analysis/SpecialFunctions/Complex/Analytic.lean (hunk 1 above)
Comment (girving) [reply Mathlib/Analysis/SpecialFunctions/Complex/Analytic.lean]: Oops, fixed.  Thanks!

**Pull request closed without merging**